src/libostree/ostree-repo-commit.c

```diff
@@ -406,6 +410,7 @@ add_size_index_to_metadata (OstreeRepo        *self,
           e_size = g_hash_table_lookup (self->object_sizes, e_checksum);
           _ostree_write_varuint64 (buffer, e_size->archived);
           _ostree_write_varuint64 (buffer, e_size->unpacked);
+          g_string_append_c(buffer, (guchar) e_size->objtype);
```

Comment:
Indeed, that would be a problem. I've reworked the compat file stuff now so that it's all internal like the detached metadata and doesn't add core objects. Let's assume that we won't be adding objects to the ostree core for now.

BTW, space before paren for consistent code style.